File: tests/fixtures/swift/sample.swift
import Foundation
import SwiftHelper

protocol ServiceProtocol {
    func start()
}

enum State {
    case ready
    case running
}

struct Config {
    let id: String
}

class SampleService: ServiceProtocol {
    let config: Config

    init(config: Config) {
        self.config = config
    }

    func start() {
        process()
    }

    func process() {
        SwiftHelper.doWork()
    }
}

extension SampleService {
    func extraMethod() {
        start()
    }
}

func globalBootstrap() {
    let service = SampleService(config: Config(id: "1"))
    service.start()
}

File: tests/fixtures/zig/sample.zig
const std = @import("std");
const helper = @import("helper.zig");

pub const Status = enum {
    Active,
    Inactive,
};

pub const User = struct {
    id: u32,
    name: []const u8,

    pub fn init(id: u32, name: []const u8) User {
        return User{
            .id = id,
            .name = name,
        };
    }

    pub fn process(self: *User) void {
        self.log();
    }

    pub fn log(self: *User) void {
        _ = self;
    }
};

pub fn runMain() void {
    helper.doWork();
}
